netbox/dcim/tables/devices.py

```diff
@@ -69,6 +62,36 @@ def get_interface_state_attribute(record):
         return "disabled"
 
 
+def get_interface_virtual_attribute(record):
+    """
+    Get interface virtual state as string to attach to <tr/> DOM element.
+    """
+    if record.is_virtual:
+        return "true"
+    else:
+        return "false"
+
+
+def get_interface_mark_connected_attribute(record):
+    """
+    Get interface enabled state as string to attach to <tr/> DOM element.
+    """
+    if record.mark_connected:
+        return "true"
+    else:
+        return "false"
+
+
+def get_interface_cable_status_attribute(record):
+    """
+    Get interface enabled state as string to attach to <tr/> DOM element.
+    """
+    if record.cable:
+        return record.cable.status
+    else:
+        return ""
+
+
```

Comment:
```suggestion
```
I think the way you are going is a little "extra" here.

Reply:
I agree, this is a lot briefer and nicer. I was trying to match the code style in the rest of the method, but if you're prefer this more brief style, hey, nobody's paying me by the line, so why not.